surpher/PactSwift
# feature: Allows custom headers for provider verification (#95)

# 📝 Summary of Changes

Changes proposed in this pull request:

- ProviderVerifier Options initialiser accepts `customHeader` argument.
- `customHeader` is a dictionary of Strings.
- Only passes key-values that contain ASCII 32-127 characters only.
- When non-allowed characters are passed they are ignored and logged into console.
- Introduces a `CharacterSet.headerASCIIAllowed` containing only allowed characters.
- Unit tests `Options` initialisation with allowed and non-allowed characters for header
- Resolves #95

# ⚠️ Items of Note

Not verified with a provider.

## 💁 Example

```swift
let providerOptions = ProviderVerifier.Options(
	provider: .init(port: 8080),
	pactsSource: .directories(["/tmp/pacts"]),
	customHeader: ["Foo": "Bar 123abc_", "Bar": "BAZ"]
)
```

## 🔨 How To Test

1. Checkout the branch `feature/provider_verification_headers_option`.
2. Set up a provider project and prepare `Options`.
3. Include `customHeader` dictionary in `Options`.
4. Run provider verification tests.
5. Custom headers should be honoured.
